src/index.ts

```diff
@@ -1,49 +1,48 @@
-import { Tracer } from '@aws-lambda-powertools/tracer';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
 import {
-  BatchGetItemCommand,
-  BatchGetItemCommandInput,
-  BatchWriteItemCommand,
-  BatchWriteItemCommandInput,
-  DeleteItemCommand,
-  DeleteItemCommandInput,
-  DynamoDBClient,
-  DynamoDBClientConfig,
-  GetItemCommand,
-  GetItemCommandInput,
-  PutItemCommand,
-  PutItemCommandInput,
-} from '@aws-sdk/client-dynamodb';
+  BatchGetCommand,
+  BatchGetCommandInput,
+  BatchWriteCommand,
+  BatchWriteCommandInput,
+  DeleteCommand,
+  DeleteCommandInput,
+  DynamoDBDocumentClient,
+  GetCommand,
+  GetCommandInput,
+  PutCommand,
+  PutCommandInput,
+} from '@aws-sdk/lib-dynamodb';
 
 export class MhDynamoDbClient {
-  private readonly client: DynamoDBClient;
+  private readonly client: DynamoDBDocumentClient;
 
-  constructor(config: DynamoDBClientConfig) {
-    const tracer = new Tracer();
-    this.client = tracer.captureAWSv3Client(new DynamoDBClient(config));
+  constructor(client?: DynamoDBClient) {
+    client = client ?? new DynamoDBClient({});
+    this.client = DynamoDBDocumentClient.from(client);
```

Comment:
Here i'm thinking the following:

- If you want to use tracer, you can always instantiate the dynamoDb client on application side and pass it here with tracer set. 
- If you don't want to use tracer but the application is in a lambda context, you still don't need the dynamodb config (credentials, region etc.) so the application don't need to pass client and we can instantiate here?

Reply:
I really think we should accept and `options` object where you can pass the `captureAWSv3Client` function. There is a middy middleware that does this exact thing `secretsManager`. I use it in `int-product-promotion-card` incoming-webhook lambda.